lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -0,0 +1,801 @@
+{
+  const Point mu(distribution_.getMean());
+  const CovarianceMatrix cov(distribution_.getCovariance());
+  const CovarianceMatrix cxx(distribution_.getMarginal(nonConditioningIndices_).getCovariance());
+  const CovarianceMatrix cyy(distribution_.getMarginal(conditioningIndices_).getCovariance());
+  const Point mux = mu.select(nonConditioningIndices_);
+  const Point muy = mu.select(conditioningIndices_);
+  Matrix cxy(conditioningIndices_.getSize(), nonConditioningIndices_.getSize());
+  for (UnsignedInteger i = 0; i < conditioningIndices_.getSize(); ++ i)
+    for (UnsignedInteger j = 0; j < nonConditioningIndices_.getSize(); ++ j)
+      cxy(i, j) = cov(conditioningIndices_[i], nonConditioningIndices_[j]);
+  // here we could get the inverse cholesky from Elliptical but we cannot access the covariance scaling factor
+  const Point muConditional(mux + cxy.transpose() * cyy.solveLinearSystem(conditioningValues_ - muy));
+  C = CovarianceMatrix((cxx - (cxy.transpose() * cyy.solveLinearSystem(cxy))).getImplementation());
+  return muConditional;
+}
+
+/* Get the simplified version */
+Bool PointConditionalDistribution::hasSimplifiedVersion(Distribution & simplified) const
+{
+  // no conditioning (empty point)
+  if (getDimension() == distribution_.getDimension())
+  {
+    simplified = distribution_;
+    return true;
+  }
+
+  // we want to disable simplification is disabled, but still include the special case above
+  if (!ResourceMap::GetAsBool("PointConditionalDistribution-UseSimplifiedVersion"))
+    return false;
+
+  // conditioning components have no influence on the other components
+  if (distribution_.hasIndependentCopula())
+  {
+    simplified = distribution_.getMarginal(nonConditioningIndices_);
+    return true;
+  }
+
+  // Normal
+  Normal *p_normal = dynamic_cast<Normal *>(distribution_.getImplementation().get());
+  if (p_normal)
+  {
+    CovarianceMatrix C;
+    const Point mu(decompose(C));
+    simplified = Normal(mu, C);
+    return true;
+  }
+
+  // Student
+  Student *p_student = dynamic_cast<Student *>(distribution_.getImplementation().get());
+  if (p_student)
+  {
+    CovarianceMatrix C;
+    const Point mu(decompose(C));
+    const Point mY(conditioningValues_ - mu.select(conditioningIndices_));
+    const Scalar dy = mY.dot(distribution_.getMarginal(conditioningIndices_).getCovariance().solveLinearSystem(mY));
+    const Scalar nu = p_student->getNu();
+    const UnsignedInteger py = conditioningIndices_.getSize();
+    C = CovarianceMatrix((C * std::sqrt((nu + dy) / (nu + py))).getImplementation());
+    Point sigma(getDimension());
+    CorrelationMatrix R(getDimension());
+    for (UnsignedInteger i = 0; i < getDimension(); ++ i)
+    {
+      const Scalar cii = C(i, i);
+      sigma[i] = std::sqrt(cii);
+      for (UnsignedInteger j = 0; j < i; ++ j)
+        R(i, j) = C(i, j) / (sigma[i] * sigma[j]);
+    }
+    simplified = Student(nu + py, mu, sigma, R);
```

Comment:
Maybe we need a Student constructor which takes C rather than (sigma, R) as input (not in this PR though)

Reply:
yes, I opened an issue for that